compiler/noirc_frontend/src/hir/resolution/import.rs

```diff
@@ -210,3 +210,18 @@ fn resolve_external_dep(
 
     resolve_path_to_ns(&dep_directive, dep_def_map, def_maps, allow_contracts)
 }
+
+pub fn get_path_crate(
+    current_crate: CrateId,
+    current_def_map: &CrateDefMap,
+    path: &Path,
+) -> Option<CrateId> {
+    match path.kind {
+        PathKind::Dep => {
+            let crate_name = &path.segments.first().unwrap();
+            current_def_map.extern_prelude.get(&crate_name.0.contents).map(|module| module.krate)
+        }
+        PathKind::Crate => Some(current_crate),
+        PathKind::Plain => Some(current_crate),
```

Comment:
Is this correct? The path could be imported from another crate. E.g:

```rs
use dep::foo;

impl foo::Trait for foo::Type {}
```

Reply:
Fix: https://github.com/noir-lang/noir/pull/2844/commits/48b191cc32bfe39896f8606cc5cf262ebb8283ed

I've slightly changed the approach, we now resolve the type and get the crate from the `StructType`